src/lib.rs

```diff
@@ -202,8 +202,13 @@ fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
         }
         Msg::SaveSelectedTodo => {
             if let Some(selected_todo) = model.selected_todo.take() {
-                if let Some(todo) = model.todos.get_mut(&selected_todo.id) {
-                    todo.title = selected_todo.title;
+                let title = selected_todo.title.trim();
+                if title.is_empty() {
+                    model.todos.remove(&selected_todo.id);
+                } else {
+                    if let Some(todo) = model.todos.get_mut(&selected_todo.id) {
+                        todo.title = title.to_owned();
+                    }
```

Comment:
Possible alternatives that can be interesting for you when you are learning Rust:

```rust
match model.selected_todo.take().as_ref().map(|todo| (todo.title.trim(), todo.id)) {
    Some((title, id)) if title.is_empty() => { model.todos.remove(&id); },
    Some((title, id)) => if let Some(todo) = model.todos.get_mut(&id) {
        todo.title = title.to_owned();
    },
    None => (),
}
```

```rust
(|| {
    let selected_todo = model.selected_todo.take()?;
    let title = selected_todo.title.trim();
    if title.is_empty() {
        model.todos.remove(&selected_todo.id);
    } else {
        model.todos.get_mut(&selected_todo.id)?.title = title.to_owned();
    }
    Some(())
})();
```
Your choice.

Reply:
Oh. The tuple `match` is fancy!